File: src/main/java/turniplabs/industry/block/BlockSolarGenerator.java
package turniplabs.industry.block;

import net.minecraft.src.EntityPlayer;
import net.minecraft.src.Material;
import net.minecraft.src.TileEntity;
import net.minecraft.src.World;
import sunsetsatellite.energyapi.EnergyAPI;
import sunsetsatellite.energyapi.template.blocks.BlockBatteryBox;
import turniplabs.industry.block.tile.TileEntitySolarGenerator;
import turniplabs.industry.container.ContainerSolarGenerator;
import turniplabs.industry.gui.GuiSolarGenerator;

public class BlockSolarGenerator extends BlockBatteryBox {
    public BlockSolarGenerator(int i, Material material) {
        super(i, material);
    }

    @Override
    public boolean blockActivated(World world, int x, int y, int z, EntityPlayer entityplayer)
    {
        if (!world.isMultiplayerAndNotHost) {
            TileEntitySolarGenerator tile = (TileEntitySolarGenerator) world.getBlockTileEntity(x, y, z);
            if (tile != null) {
                EnergyAPI.displayGui(entityplayer, new GuiSolarGenerator(entityplayer.inventory, tile), new ContainerSolarGenerator(entityplayer.inventory, tile), tile);
            }
        }
        return true;
    }

    @Override
    protected TileEntity getBlockEntity() {
        return new TileEntitySolarGenerator();
    }

}

File: src/main/kotlin/turniplabs/industry/block/machine/BlockSolarGenerator.kt
package turniplabs.industry.block.machine

import net.minecraft.src.*
import sunsetsatellite.energyapi.EnergyAPI
import turniplabs.industry.block.tile.TileEntitySolarGenerator
import turniplabs.industry.gui.ContainerSolarGenerator
import turniplabs.industry.gui.GuiSolarGenerator

class BlockSolarGenerator(i: Int, material: Material?) : BlockContainer(i, material) {

    override fun getBlockEntity(): TileEntity {
        return TileEntitySolarGenerator()
    }

    override fun blockActivated(world: World?, x: Int, y: Int, z: Int, entityPlayer: EntityPlayer?): Boolean {
        if (!world!!.isMultiplayerAndNotHost) {
            val tile: TileEntitySolarGenerator = world.getBlockTileEntity(x, y, z) as TileEntitySolarGenerator

            if (tile != null)
                EnergyAPI.displayGui(entityPlayer, GuiSolarGenerator(entityPlayer!!.inventory, tile),
                    ContainerSolarGenerator(entityPlayer.inventory, tile), tile)
        }
        return true
    }

    override fun onBlockRemoval(world: World?, x: Int, y: Int, z: Int) {
        val inventory: IInventory = world?.getBlockTileEntity(x, y, z) as IInventory

        label0@
        for (invSize in 0 until inventory.sizeInventory) {
            val itemStack: ItemStack = inventory.getStackInSlot(invSize) ?: continue

            val fX: Float = (world.rand.nextFloat() * 0.8f) + 0.1f
            val fY: Float = (world.rand.nextFloat() * 0.8f) + 0.1f
            val fZ: Float = (world.rand.nextFloat() * 0.8f) + 0.1f
            do {
                if (itemStack.stackSize <= 0)
                    continue@label0

                var worldRand: Int = world.rand.nextInt(21) + 10
                if (worldRand > itemStack.stackSize)
                    worldRand = itemStack.stackSize
                itemStack.stackSize -= worldRand

                val item = EntityItem(
                    world,
                    (x.toFloat() + fX).toDouble(),
                    (y.toFloat() + fY).toDouble(),
                    (z.toFloat() + fZ).toDouble(),
                    ItemStack(itemStack.itemID, worldRand, itemStack.metadata, itemStack.tag)
                )
                item.motionX = world.rand.nextGaussian() * 0.05f
                item.motionY = world.rand.nextGaussian() * 0.05f
                item.motionZ = world.rand.nextGaussian() * 0.05f
                world.entityJoinedWorld(item)
            } while (true)
        }
        super.onBlockRemoval(world, x, y, z)
    }
}